### PR TITLE
chore: replace blake2b implementation by golang.org/x/crypto

### PR DESCRIPTION
**Pull request by Jorropo**
In 2016-12-13 d8e61c69ab46ca38328da2f4995abaf93b252290 golang.org/x/crypto gained a blake2b AVX implementation.

There is also an AVX2 one.

It is faster than what we are using right now.

This commit allows for faster code and remove a dependency.

For future improvements (like NEON) /x/crypto seems more maintained (it isn't an archived repo).

```
benchmark              old ns/op     new ns/op     delta
BenchmarkSum128-12     252           140           -44.35%
BenchmarkSum1K-12      1221          986           -19.24%

benchmark              old MB/s     new MB/s     speedup
BenchmarkSum128-12     507.37       911.98       1.80x
BenchmarkSum1K-12      838.70       1038.43      1.24x
```

### CODE REVIEW

File: version.json
Comment (marten-seemann) [inline version.json:2]: This probably needs to be v0.2.0, as you're dropping a dependency.

File: register/blake2/multihash_blake2.go
Comment (marten-seemann) [inline register/blake2/multihash_blake2.go:31]: Nit: drop the blake2 here, this function is not doing anything blake-specific. Maybe `hasherMust`?

**Review by marten-seemann — COMMENTED** [review]
This looks good to me, but I'll leave the final approval to someone who has more context with these hash functions.

File: version.json
Comment (Jorropo) [reply version.json:2]: The CI disagree. 🙂 

https://github.com/multiformats/go-multihash/runs/6705660054?check_suite_focus=true

Comment (marten-seemann) [reply version.json:2]: It does not:
```
`gorelease` says:
  ```
  # summary
  Suggested version: v0.2.0
  ```
```

File: register/blake2/multihash_blake2.go
Comment (Jorropo) [reply register/blake2/multihash_blake2.go]: Removing this function all together.

**Review by marten-seemann — APPROVED** [review]

**Comment by aschmahmann** [comment]
@Kubuxu any issues with merging?

@Jorropo not blocking here, but it'd be nice if there was some verification that the hashes from the different blake2 libraries are the same. AFAICT we don't have any here, either a Go test or adding to https://github.com/multiformats/multihash/ and bumping the dependency here would be great.

**Comment by marten-seemann** [comment]
> not blocking here, but it'd be nice if there was some verification that the hashes from the different blake2 libraries are the same

That would need to live in a separate Go module though, otherwise the dependency would stay in go.mod. Or maybe a one-time test would be enough to convince us that the libraries are doing the same?

**Comment by Jorropo** [comment]
I'm surprised there is no blake2 tests, Ill ad some, likely blake2's official test set.

**Comment by aschmahmann** [comment]
> That would need to live in a separate Go module though, otherwise the dependency would stay in go.mod. Or maybe a one-time test would be enough to convince us that the libraries are doing the same?

I was thinking that if you did a PR with the tests separately from this one we could merge that, and then rebase this PR on top of master and make sure if passes. Since the tests would be hitting test vectors that are constant it'd be fine.

**Comment by Jorropo** [comment]
I dont think we actually need to check boths, blake2 is a spec if an implementation isnt following it its wrong and that likely a big security issue.

The tests can compare to well known values.

**Comment by Jorropo** [comment]
/cc @aschmahmann I have added blake2b tests, tested both on the old and new implementation they both work.

RFM on my side

**Comment by Jorropo** [comment]
> I trust golang/x/crypto to implement Blake2 correctly.

Fair enough, I belive the goal was more to ensure the new registering is done correctly.
Maybe one used to take bits now it takes byte sizes, or something weird like that.

**Comment by Kubuxu** [comment]
Hey, tests seem to fail on the CI.
Sorry for the delay, I didn't know you were waiting on me.

**Comment by Jorropo** [comment]
@Kubuxu 
Sorry for the tests I had fixed it locally it but forgot to push it.

I don't know if it's waiting on you, you are just the only one with merge permissions in the conversation afait.

**Comment by Jorropo** [comment]
Thx, sorry for pinging you Kubuxu, weird that Marten only show up as Contributor.

**Pull request merged**
